paasta_tools/cli/cmds/wait_for_deployment.py

```diff
@@ -0,0 +1,323 @@
+            "are required for this command to work."
+        )
+    )
+    list_parser.add_argument(
+        '-c', '-k', '--commit',
+        help='Git sha to wait for deployment',
+        required=True,
+        type=validate_full_git_sha,
+    )
+    list_parser.add_argument(
+        '-l', '--deploy-group',
+        help='deploy group (e.g. cluster1.canary, cluster2.main).',
+        required=True,
+    ).completer = lazy_choices_completer(list_deploy_groups)
+    list_parser.add_argument(
+        '-s', '--service',
+        help='Name of the service which you wish to wait for deployment. '
+        'Leading "services-" will be stripped.',
+        required=True,
+    ).completer = lazy_choices_completer(list_services)
+    list_parser.add_argument(
+        '-t', '--timeout',
+        dest="timeout",
+        type=int,
+        default=DEFAULT_DEPLOYMENT_TIMEOUT,
+        help=(
+            "Time in seconds to wait for paasta to deploy the service. "
+            "If the timeout is exceeded we return 1. "
+            "Default is %(default)s seconds."
+        ),
+    )
+    list_parser.add_argument(
+        '-d', '--soa-dir',
+        dest="soa_dir",
+        metavar="SOA_DIR",
+        default=DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+    list_parser.add_argument(
+        '-v', '--verbose',
+        action='count',
+        dest="verbose",
+        default=0,
+        help="Print out more output."
+    )
+
+    list_parser.set_defaults(command=paasta_wait_for_deployment)
+
+
+def paasta_wait_for_deployment(args):
+    """Wrapping wait_for_deployment"""
+    if args.verbose:
+        log.setLevel(level=logging.DEBUG)
+    else:
+        log.setLevel(level=logging.INFO)
+
+    service = args.service
+    if service and service.startswith('services-'):
+        service = service.split('services-', 1)[1]
+    try:
+        validate_service_name(service, soa_dir=args.soa_dir)
+    except NoSuchService as e:
+        print(PaastaColors.red('%s' % e))
+        return 1
+
+    in_use_deploy_groups = list_deploy_groups(service=service,
+                                              soa_dir=args.soa_dir)
+    _, invalid_deploy_groups = \
+        validate_given_deploy_groups(in_use_deploy_groups, [args.deploy_group])
+
+    if len(invalid_deploy_groups) == 1:
+        print(PaastaColors.red("ERROR: These deploy groups are not currently "
+                               "used anywhere: %s.\n" %
+                               (",").join(invalid_deploy_groups)))
+        print(PaastaColors.red("You probably need one of these in-use deploy "
+                               "groups?:\n   %s" %
+                               (",").join(in_use_deploy_groups)))
+        return 1
+
+    old_git_sha = get_currently_deployed_sha(service=service,
+                                             deploy_group=args.deploy_group)
+    if old_git_sha == args.commit:
```

Comment:
I'm not sure this error message really makes in this context. It will pretty much *always* be the case that you are waiting on what is set to be deployed.

This needs to be only in mark_for_deployment